packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.ts

```diff
@@ -70,23 +76,31 @@ export class CryptoWatchPriceFeed extends PriceFeedInterface {
   }
 
   public getCurrentPrice(): null | BN {
-    if (!this.twapLength && this.invertPrice) {
```

Comment:
Reading `this.twapLength` is only neccessary in this method because we compute a TWAP in `_inversePriceSafely`, which is non-intuitive. I've changed this logic so we no longer need to read this here. This also is a problem to this PR because `update` now reads `twapLength` potentially from `ancillaryData`, but does not set the class's state variable `this.twapLength`.